zigpy_xbee/zigbee/application.py

```diff
@@ -129,9 +129,12 @@ def handle_rx(self, src_ieee, src_nwk, src_ep, dst_ep, cluster_id, profile_id, r
 
         ember_ieee = zigpy.types.EUI64(src_ieee)
         if ember_ieee not in self.devices:
-            self.handle_join(src_nwk, ember_ieee, 0)  # TODO: Parent nwk
-        self._devices_by_nwk[src_nwk] = src_ieee
+            self.handle_join(src_nwk, ember_ieee, 0)  # TODO: Parent nwk       
         device = self.get_device(ember_ieee)
+        if device.nwk != src_nwk:
```

Comment:
I think it is better to conflate line 131 if block and line 134 if block. IEEE never changes, but `if src_nwk not in self._devices_by_nwk` then either NWK has changed or it is a new device. In both cases we should `self.handle_join()`.

Reply:
If statements conflated.  Do we still want to remove the original nwk?